conftest.py

```diff
@@ -89,22 +90,46 @@ def small_cluster(request):
         yield cluster
 
 
+# this code was taken from pytest docs
+# https://docs.pytest.org/en/latest/example/simple.html#making-test-result-information-available-in-fixtures
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    # execute all other hooks to obtain the report object
+    outcome = yield
+    rep = outcome.get_result()
+
+    # set a report attribute for each phase of a call, which can
+    # be "setup", "call", "teardown"
+
+    setattr(item, "rep_" + rep.when, rep)
+
+
 @pytest.fixture
-def small_client(small_cluster):
+def small_client(small_cluster, s3_cluster_dump_url, s3_storage_options, request):
     with Client(small_cluster) as client:
         small_cluster.scale(10)
         client.wait_for_workers(10)
         client.restart()
+
         yield client
 
+        cluster_dump = strtobool(os.environ.get("CLUSTER_DUMP", "false"))
+
+        if cluster_dump and request.node.rep_call.failed:
+            dump_path = os.path.join(s3_cluster_dump_url, small_cluster.name)
+            print(f"Cluster state dump can be found at: {dump_path}")
+            client.dump_cluster_state(dump_path, **s3_storage_options)
+
 
 S3_REGION = "us-east-2"
 S3_BUCKET = "s3://coiled-runtime-ci"
 
 
 @pytest.fixture(scope="session")
 def s3_storage_options():
-    return {"config_kwargs": {"region_name": S3_REGION}}
+    # add "auto_mkdir": False until this is fixed
+    # https://github.com/fsspec/filesystem_spec/issues/974
+    return {"config_kwargs": {"region_name": S3_REGION, "auto_mkdir": False}}
```

Comment:
@gjoseph92 Maybe I'm doing something wrong, I tried this as you suggested but I'm getting a TypeError https://github.com/coiled/coiled-runtime/runs/6712573736?check_suite_focus=true#step:6:167

Reply:
I believe you want to add `"auto_mkdir": False` at the same level as `config_kwargs`. Something like (note I've not tested the code suggestion below):

```suggestion
    return {"config_kwargs": {"region_name": S3_REGION}, "auto_mkdir": False}
```